translate/src/modules/machinery/components/source/GoogleTranslation.tsx

```diff
@@ -38,82 +38,70 @@ export function GoogleTranslation({
-          locale.name,
-        );
+  const handleTransformation = async (characteristic: string) => {
+    // Only fetch transformation if not reverting to original
+    if (characteristic !== 'original') {
+      const machineryTranslations = await fetchGPTTransform(
+        translation.original,
+        currentTranslation,
+        characteristic,
+        locale.name,
+      );
 
-        if (machineryTranslations.length > 0) {
-          const translationWithoutQuotes =
-            machineryTranslations[0].translation.replace(
-              /^['"](.*)['"]$/,
-              '$1',
-            );
-          setCurrentTranslation(translationWithoutQuotes);
-          setLLMTranslation(translationWithoutQuotes);
-          setShowOriginalOption(true);
-        }
-      } else {
-        console.log(translation.translation);
-        setCurrentTranslation(translation.translation);
-        setLLMTranslation('');
-        setSelectedOption('');
-        setShowOriginalOption(false);
+      if (machineryTranslations.length > 0) {
+        setCurrentTranslation(machineryTranslations[0].translation);
+        setLLMTranslation(machineryTranslations[0].translation);
+        setShowOriginalOption(true);
       }
-    } catch (error) {
-      console.error('Error fetching GPT transformation:', error);
+    } else {
+      setCurrentTranslation(translation.translation);
+      setLLMTranslation('');
+      setSelectedOption('');
+      setShowOriginalOption(false);
     }
   };
 
-  const handleOptionClick = (ev: React.MouseEvent, option: string) => {
+  const handleOptionClick = (ev: React.MouseEvent<HTMLLIElement>) => {
     ev.stopPropagation();
-    setSelectedOption(option);
-    setDropdownOpen(false);
+    const target = ev.currentTarget;
+    const characteristic = target.dataset['characteristic'];
 
-    let characteristic = '';
-    switch (option) {
-      case 'Rephrase':
-        characteristic = 'alternative';
-        setSelectedOption('REPHRASED');
-        break;
-      case 'Make formal':
-        characteristic = 'formal';
-        setSelectedOption('FORMAL');
-        break;
-      case 'Make informal':
-        characteristic = 'informal';
-        setSelectedOption('INFORMAL');
-        break;
-      case 'Show original':
-        characteristic = 'original'; // Special handling to revert to original
-        break;
-      default:
-        break;
-    }
+    if (characteristic) {
+      let displayText = '';
 
-    const trackllmTranslation = option;
-    onLLMTranslationChange(trackllmTranslation);
+      switch (characteristic) {
+        case 'alternative':
+          displayText = 'REPHRASED';
+          break;
+        case 'formal':
+          displayText = 'FORMAL';
+          break;
+        case 'informal':
+          displayText = 'INFORMAL';
+          break;
+        case 'original':
+          displayText = '';
+          break;
+        default:
+          break;
+      }
+      setSelectedOption(displayText);
```

Comment:
If the dropdown options are localised, the selected value display should also be localised. But this is messy enough already, so I'd be fine with a "TODO" or "FIXME" code comment here to that effect.